refactor: dropping `ExtendedNote`

benesjan opened the pull request:
When checking how note filters are used I noticed that the ExtendedNote type is no longer used so I dropped it in this PR.

benesjan commented:
* **#17868** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17868?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17868?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

wildjos commented on yarn-project/stdlib/src/note/unique_note.ts:
```diff
@@ -143,7 +78,11 @@ export class UniqueNote extends ExtendedNote {
     return new this(note, recipient, contractAddress, storageSlot, txHash, noteNonce);
```
Just a minor point - `return new this(...)` looks like a leftover from when UniqueNote extended ExtendedNote. Since it’s no longer subclassed, maybe change to `new UniqueNote(...)`

wildjos commented:
I noticed there’s no equals() method on UniqueNote, so the tests rely on Jest’s toEqual.
What are your thoughts on adding an explicit equals() to the class?

wildjos reviewed (approved):
Looks good — I left a couple of comments for you to address :)

benesjan commented:
> I noticed there’s no equals() method on UniqueNote, so the tests rely on Jest’s toEqual.
> What are your thoughts on adding an explicit equals() to the class?

We always just implemented these when needed so I would not bother with this here.

benesjan committed:
refactor: dropping `ExtendedNote`

When checking how note filters are used I noticed that the ExtendedNote type is no longer used so I dropped it in this PR.
